### PR TITLE
Fix inserting defaults for missing values in columnar formats

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix inserting defaults for missing values in columnar formats. Previously missing columns were filled with defaults for types, not for columns.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Storages/StorageFile.cpp [hunk 1]
```diff
@@ -691,8 +691,7 @@ Pipe StorageFile::read(
         const auto get_columns_for_format = [&]() -> ColumnsDescription
         {
             if (isColumnOriented())
-                return ColumnsDescription{
-                    storage_snapshot->getSampleBlockForColumns(column_names).getNamesAndTypesList()};
```
Comment (Avogar) [inline src/Storages/StorageFile.cpp:695]: Bug was here, such `ColumnsDescription` didn't contain any information about default expressions for columns.

File: src/Storages/StorageSnapshot.cpp [hunk 2]
```diff
@@ -124,6 +124,39 @@ Block StorageSnapshot::getSampleBlockForColumns(const Names & column_names) cons
...
+        }
+        else
+        {
+            throw Exception(ErrorCodes::NOT_FOUND_COLUMN_IN_BLOCK,
+                            "Column {} not found in table {}", backQuote(name), storage.getStorageID().getNameForLogs());
+        }
+    }
+
+    return res;
+}
```
Comment (tavplubix) [inline src/Storages/StorageSnapshot.cpp:158]: Is it possible to avoid copy-paste?

File: src/Storages/StorageURL.cpp [hunk 3]
```diff
@@ -602,8 +602,7 @@ Pipe IStorageURLBase::read(
     Block block_for_format;
     if (isColumnOriented())
     {
-        columns_description = ColumnsDescription{
-            storage_snapshot->getSampleBlockForColumns(column_names).getNamesAndTypesList()};
+        columns_description = storage_snapshot->getDescriptionForColumns(column_names);
```
Comment (tavplubix) [inline src/Storages/StorageURL.cpp:605]: What about `StorageURLWithFailover::read(...)`?

**Review by tavplubix — APPROVED** [review]

File: src/Storages/StorageSnapshot.cpp (hunk 2 above)
Comment (Avogar) [reply src/Storages/StorageSnapshot.cpp]: Done

File: src/Storages/StorageURL.cpp (hunk 3 above)
Comment (Avogar) [reply src/Storages/StorageURL.cpp:605]: Didn't noticed it, fixed. Thanks!

**Comment by tavplubix** [comment]
Stateless tests (thread, actions) [3/3] - failure is interesting - #37452

**Comment by Avogar** [comment]
> Stateless tests (thread, actions) [3/3] - failure is interesting - https://github.com/ClickHouse/ClickHouse/issues/37452
Oh, it will fail in master now, my fault. I can do fast fix to prevent red test in master (didn't understand quickly how to fix data race)

**Pull request merged**